File: quiz_application_Back/src/main/java/com/quiz/service/QuizService.java
package com.quiz.service;

import java.util.List;
import java.util.Set;

import com.quiz.dto.QuizDto;
import com.quiz.entity.Category;
import com.quiz.entity.Quiz;

public interface QuizService {
	
	    public QuizDto addQuiz(QuizDto quiz); 
	    
	    public  QuizDto updateQuiz(QuizDto quiz,long qId);
	    
	    public Set<QuizDto> getQuizzes();

	    public  QuizDto getQuiz(Long quizId);
	    
	    public  void  delteQuiz(Long quizId);
	    
	  
	    public List<QuizDto> getQuizzesOfCategory(Category category);

	    public  List<QuizDto> getActiveQuizzess();
	    public  List<QuizDto> getActiveQuizzessOfCategory(Category c);

}

File: quiz_application_front/src/app/services/quiz.service.ts
import { Injectable } from '@angular/core';
import baseUrl from './helper';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Quiz } from '../Classes/quiz';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private _http: HttpClient) { }
  public quizzes():Observable<Quiz[]>{
    return this._http.get<Quiz[]>(`${baseUrl}/quiz/`)
  }

  //add quiz
  public addQuiz(quiz : Quiz):Observable<Object>{
   return this._http.post(`${baseUrl}/quiz/`,quiz);
      
  }

  //delte quiz
  public deleteQuiz(qid: Number):Observable<object>{
    return this._http.delete(`${baseUrl}/quiz/${qid}`);
  }

  //get the single quiz

  public getQuiz(qId :number):Observable<Quiz>{
    return this._http.get<Quiz>(`${baseUrl}/quiz/${qId}`);
  }

  public updateQuiz(quizId:number,quiz:Quiz):Observable<object>{
    return this._http.put(`${baseUrl}/quiz/${quizId}`,quiz);
  }

  //get Quizzes of category
  public getQuizzesOfCategory(cId:any){
    return this._http.get(`${baseUrl}/quiz/category/${cId}`)
  }


  // get Active quizzess
  public getActiveQuizzes():Observable<Quiz[]>{
    return this._http.get<Quiz[]>(`${baseUrl}/quiz/active`);
  }

  //get active quizzes of category
  public getActiveQuizzesofCategory(cId:any):Observable<Quiz[]>{
    return this._http.get<Quiz[]>(`${baseUrl}/quiz/category/active/${cId}`);
  }
}
